lib/pwwka/persistent_transmitter.rb

```diff
@@ -0,0 +1,73 @@
+require_relative "publish_options"
+
+begin  # optional dependency
+  require 'resque'
+  require 'resque-retry'
+rescue LoadError
+end
+
+module Pwwka
+  # Primary interface for sending messages.
+  #
+  # Example:
+  #
+  #     # Send a message, blowing up if there's any problem
+  #     Pwwka::Transmitter.send_message!({ user_id: @user.id }, "users.user.activated")
+  #
+  #     # Send a message, logging if there's any problem
+  #     Pwwka::Transmitter.send_message_safely({ user_id: @user.id }, "users.user.activated")
+  class PersistentTransmitter
+
+    extend Pwwka::Logging
+    include Pwwka::Logging
+
+    DEFAULT_DELAY_BY_MS = 5000
+
+    attr_reader :channel_connector
+    private_class_method :new
+
+    def self.batch(routing_key, type: nil, headers: nil, message_id: :auto_generate)
+      transmitter = self.new
+      yeild(transmitter)
+    ensure
+      transmitter.channel_connector.connection_close
+    end
+
+    def initialize
+      @channel_connector = ChannelConnector.new(connection_name: "p: #{Pwwka.configuration.app_id} #{Pwwka.configuration.process_name}".strip)
```

Comment:
nothing for now, but something to think about going forward.  I'm pretty sure you don't want to share a channel across multiple threads so we may need to separate the connection from the channel in future iterations.